### PR TITLE
Add regular logs to filters straight from BlockAddedEvent. Still use `privacyQueries` for private logs

### PR DESCRIPTION
**Pull request by RatanRSur**
## PR description
@pinges correctly pointed out that, even though we have access to all the logs in the `BlockAddedEvent`, we still use `BlockchainQueries` to add the new logs from the event. Now we do the sane thing.

Unfortunately, it doesn't seem as easy with private logs so we still have to use `privacyQueries` as far as I can tell.

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java [hunk 1]
```diff
@@ -151,28 +152,28 @@ public boolean uninstallFilter(final String filterId) {
     }
   }
 
-  public void recordBlockEvent(final BlockAddedEvent event, final Blockchain blockchain) {
+  public void recordBlockEvent(final BlockAddedEvent event, final Blockchain __) {
```
Comment (davemec) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java:155]: I see that it is not being used by the method, but why the underscore variable name? Is it just how you are denoting that it is not used?

Comment (RatanRSur) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java:155]: Yup! Common functional pattern to let the reader know early that they don't have to worry about it.

Comment (davemec) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java:155]: Should we handle that at the lambda level and for the method signature actually just remove the param?

Comment (RatanRSur) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java:155]: If we never use it then for sure, I'll have to look into that.

Comment (RatanRSur) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java:156]: You were right, it was pretty useless. It was only used in one place that already had access to the blockchain in the scope anyways!

Comment (RatanRSur) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java:156]: I made a separate PR removing it

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java [hunk 2]
```diff
@@ -151,28 +153,45 @@ public boolean uninstallFilter(final String filterId) {
...
-
-    final List<LogWithMetadata> logs = findLogsWithinRange(filter, fromBlockNumber, toBlockNumber);
-
-    filter.addLogs(logs);
+    final List<LogWithMetadata> logsWithMetadata = event.getLogsWithMetadata();
+    filterRepository.getFiltersOfType(LogFilter.class).stream()
+        .filter(
+            // Only keep filters where the "to" block could include the block in the event
+            filter -> {
+              final OptionalLong maybeToBlockNumber = filter.getToBlock().getNumber();
```
[the remark this answers is not in the document]
Comment (RatanRSur) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java:171]: I'm not sure if I understand but these `LogFilter`s don't specify a block hash.

[the remark this answers is not in the document]
Comment (lucassaldanha) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java:171]: Ah nvm... I mixed up with eth_getLogs!

**Pull request merged**